gluoncv/data/mhp.py

```diff
@@ -0,0 +1,182 @@
+"""Multi-Human-Parsing V1 Dataset."""
+import os
+import numpy as np
+from PIL import Image
+from PIL import ImageFile
+
+import mxnet as mx
+from .segbase import SegmentationDataset
+
+ImageFile.LOAD_TRUNCATED_IMAGES = True
+
+
+class MHPV1Segmentation(SegmentationDataset):
+    """Multi-Human-Parsing V1 Dataset.
+    Parameters
+    ----------
+    root : string
+        Path to VOCdevkit folder. Default is '$(HOME)/mxnet/datasplits/mhp/LV-MHP-v1'
+    split: string
+        'train', 'val' or 'test'
+    transform : callable, optional
+        A function that transforms the image
+    Examples
+    --------
+    >>> from mxnet.gluon.data.vision import transforms
+    >>> # Transforms for Normalization
+    >>> input_transform = transforms.Compose([
+    >>>     transforms.ToTensor(),
+    >>>     transforms.Normalize([.485, .456, .406], [.229, .224, .225]),
+    >>> ])
+    >>> # Create Dataset
+    >>> trainset = gluoncv.data.MHPV1Segmentation(split='train', transform=input_transform)
+    >>> # Create Training Loader
+    >>> train_data = gluon.data.DataLoader(
+    >>>     trainset, 4, shuffle=True, last_batch='rollover',
+    >>>     num_workers=4)
+    """
+    # pylint: disable=abstract-method
+    NUM_CLASS = 18
+
+    def __init__(self, root=os.path.expanduser('~/.mxnet/datasets/mhp/LV-MHP-v1'),
+                 split='train', mode=None, transform=None, **kwargs):
+        super(MHPV1Segmentation, self).__init__(root, split, mode, transform, **kwargs)
+        assert os.path.exists(root), "Please setup the dataset using" + "scripts/datasets/mhp_v1.py"
+        self.images, self.masks = _get_mhp_pairs(root, split)
+        assert (len(self.images) == len(self.masks))
+        if len(self.images) == 0:
+            raise(RuntimeError("Found 0 images in subfolders of: \
+                " + root + "\n"))
+
+    def __getitem__(self, index):
+        img = Image.open(self.images[index]).convert('RGB')
+
+        # nan check
+        img_np = np.array(img, dtype=np.uint8)
+        assert not np.isnan(np.sum(img_np))
+
+        if self.mode == 'test':
+            img = self._img_transform(img)
+            if self.transform is not None:
+                img = self.transform(img)
+            return img, os.path.basename(self.images[index])
+
+        mask = _get_mask(self.masks[index])
+
+        # algin data
+        w, h = img.size
+        if h < w:
+            oh = 768
```

Comment:
768 should be a param that passed in. Usually we don't hardcode numbers inside a function.